pkg/tsdb/influxdb/response_parser.go

```diff
@@ -86,14 +88,25 @@ func transformRows(rows []Row, query *Query) data.Frames {
 			// set a nice name on the value-field
 			valueField.SetConfig(&data.FieldConfig{DisplayNameFromDS: name})
 
-			frames = append(frames, data.NewFrame(name, timeField, valueField))
+			frames = append(frames, newDataFrame(name, query.RawQuery, *timeField, *valueField))
 		}
 	}
 
 	return frames
 }
 
-func formatFrameName(row Row, column string, query *Query) string {
+func newDataFrame(name string, queryString string, timeField data.Field, valueField data.Field) *data.Frame {
```

Comment:
this is about pointers-or-not-pointers.
my understanding of go is, that in general, prefer not using pointers, and only switch to them when it is obviously required. so from that point of view, i'm ok with `timeField` and `valueField` not being pointers.

but, in this case, at the end they will be used only as pointers in the line below,
so i recommend changing it so that `newDataFrame`'s signature becomes:

```go
func newDataFrame(name string, queryString string, timeField *data.Field, valueField *data.Field) *data.Frame {}
```